06x-new-kafka-roller.md

```diff
@@ -0,0 +1,331 @@
+```
+         nodeRef: combined-3/3
+         nodeRoles: controller,broker
+         state: RESTARTED
+         lastTransition: 654987
+         reason: POD_UNRESPONSIVE
+         numRestartAttempts: 1
+         numReconfigAttempts: 0
+         numRetries: 0
+```
+3. `combined-3` state becomes `SERVING` and since its `numRestartAttempts` is greater than 1, the roller checks the rest of the nodes. 
+4. The roller checks which node is the active controller and finds that `controller-0` is. It then sends a request to the active controller via AdminClient to describe its `controller.quorum.fetch.timeout` config value.
+5. It then considers restarting `controller-1` and checks if the quorum health would be impacted. The operator sends a request to to the active controller to describe the quorum replication state. It finds that majority of the follower controllers have caught up with the quorum leader within the `controller.quorum.fetch.timeout.ms`. 
+6. The roller restarts `controller-1` as it has no impact on the quorum health. When it has `SERVING` state, the roller repeats the quorum check and restarts `controller-2` and then `controller-0`. 
+7. It then considers restarting `combined-4`, so it performs quorum healthcheck and then availability check. Both check passes therefore `combined-4` is restarted. The same is repeated for `combined-5`.
+8. All controller and combined nodes have `SERVING` state and `numRestartAttempts` set to 1, therefore the roller checks the broker nodes.
+9. It sends a request to describe all the topic partitions, and the following list of topics is returned:
+```
+topic("topic-A"), Replicas(9, 10, 11), ISR(9, 10)
+topic("topic-B"), Replicas(6, 7, 8), ISR(6, 7, 8)
+topic("topic-C"), Replicas(10, 8, 6), ISR(10, 8, 6)
+topic("topic-D"), Replicas(7, 9, 11), ISR(7, 9, 11)
+topic("topic-E"), Replicas(6, 10, 11), ISR(6, 10, 11)
+```
+They are configured with `min.insync.replica` of 2.
+
+10. The roller batches the nodes that do not have any topic partition in common and the following batches are created: 
+- (11, 8) - `broker-11` and `broker-8` do not share any topic partitions.
+- (7) - `broker-7` and `broker-10` do not share any topic partitions, however topic-A is at min ISR, therefore 10 cannot be restarted and is removed from the batch.
+- (6) - `broker-6` and `broker-9` do not share any topic partitions, however topic-A is at min ISR, therefore 9 cannot be restarted and is removed from the batch.
+
+11. The roller picks the largest batch containing `broker-11` and `broker-8` and restarts them. It waits forthe nodes to have `SERVING` and then `LEADING_ALL_PREFERRED` state.
+12. It then restarts the batch containing only `broker-7` and restart it. It waits for it to have `SERVING` and then `LEADING_ALL_PREFERRED` state.
+13. It then restarts the batch containing only `broker-6` and restart it. It times out waiting for it to have `SERVING` state because it's still performing log recovery.
```

Comment:
```suggestion
13. It then restarts the batch containing only `broker-6`. It times out waiting for it to have `SERVING` state because it's still performing log recovery.
```